src/nvim/option.c

```diff
@@ -1700,33 +1689,33 @@ static void didset_options2(void)
 /// Check for string options that are NULL (normally only termcap options).
 void check_options(void)
 {
-  for (int opt_idx = 0; options[opt_idx].fullname != NULL; opt_idx++) {
+  for (OptIndex opt_idx = 0; options[opt_idx].fullname != NULL; opt_idx++) {
```

Comment:
I think this change should be reverted, because `OptIndex` doesn't include the index of the final `NULL` option.

Reply:
@zeertzjq Instead of reverting it, I instead added a `kOptIndexCount` and removed the `NULL` option. That way we can use `kOptIndexCount` to iterate through the `options[]` array instead.